NominatimUpdater fails with "java.sql.SQLException: No suitable driver"

I've sucessfully set up our Photon instance and the initial import worked without a hitch.

However, when I try to update the data, I see the following error instead:

```
2020-06-02 10:48:55,362 [Thread-58] INFO  de.komoot.photon.nominatim.NominatimUpdater - Starting rank 1
Exception in thread "Thread-58" org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection; nested exception is java.sql.SQLException: Cannot create JDBC driv
        at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:82)
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:612)
        at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:669)
        at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:700)
        at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:712)
        at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:763)
        at org.springframework.jdbc.core.JdbcTemplate.queryForList(JdbcTemplate.java:844)
        at de.komoot.photon.nominatim.NominatimUpdater.getIndexSectors(NominatimUpdater.java:147)
        at de.komoot.photon.nominatim.NominatimUpdater.update(NominatimUpdater.java:54)
        at de.komoot.photon.App.lambda$startApi$1(App.java:219)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.sql.SQLException: Cannot create JDBC driver of class 'org.postgis.jts.JtsWrapper' for connect URL 'jdbc:postgresql://nominatim:5432/nominatim'
        at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:75)
        at org.apache.commons.dbcp2.BasicDataSource.createConnectionFactory(BasicDataSource.java:472)
        at org.apache.commons.dbcp2.BasicDataSource.createDataSource(BasicDataSource.java:538)
        at org.apache.commons.dbcp2.BasicDataSource.getConnection(BasicDataSource.java:753)
        at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:158)
        at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:116)
        at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:79)
        ... 10 more
Caused by: java.sql.SQLException: No suitable driver
        at org.apache.commons.dbcp2.DriverFactory.createDriver(DriverFactory.java:68)
        ... 16 more

```
I investigated a little and saw that the JDBC datasource's connection string are constructed a little differently in  and `NominatimConnector` and `NominatimUpdater`.

- https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java#L238
- https://github.com/komoot/photon/blob/master/src/main/java/de/komoot/photon/nominatim/NominatimUpdater.java#L189

It looks like the `_jts`suffix is required. 

Wouldn't a reusable method to construct the datasource more suitable?

--- a/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java
+++ b/src/main/java/de/komoot/photon/nominatim/NominatimConnector.java
@@ -233,6 +233,13 @@ public class NominatimConnector {
      * @param password db username's password
      */
     public NominatimConnector(String host, int port, String database, String username, String password) {
+        BasicDataSource dataSource = buildDataSource(host, port, database, username, password);
+
+        template = new JdbcTemplate(dataSource);
+        template.setFetchSize(100000);
+    }
+
+    static BasicDataSource buildDataSource(String host, int port, String database, String username, String password) {
         BasicDataSource dataSource = new BasicDataSource();
 
         dataSource.setUrl(String.format("jdbc:postgres_jts://%s:%d/%s", host, port, database));
@@ -240,9 +247,7 @@ public class NominatimConnector {
         dataSource.setPassword(password);
         dataSource.setDriverClassName(JtsWrapper.class.getCanonicalName());
         dataSource.setDefaultAutoCommit(false);
-
-        template = new JdbcTemplate(dataSource);
-        template.setFetchSize(100000);
+        return dataSource;
     }
 
     public void setImporter(Importer importer) {

--- a/src/main/java/de/komoot/photon/nominatim/NominatimUpdater.java
+++ b/src/main/java/de/komoot/photon/nominatim/NominatimUpdater.java
@@ -184,13 +184,7 @@ public class NominatimUpdater {
      * @param password Nominatim database password
      */
     public NominatimUpdater(String host, int port, String database, String username, String password) {
-        BasicDataSource dataSource = new BasicDataSource();
-
-        dataSource.setUrl(String.format("jdbc:postgresql://%s:%d/%s", host, port, database));
-        dataSource.setUsername(username);
-        dataSource.setPassword(password);
-        dataSource.setDriverClassName(JtsWrapper.class.getCanonicalName());
-        dataSource.setDefaultAutoCommit(true);
+        BasicDataSource dataSource = NominatimConnector.buildDataSource(host, port, database, username, password);
 
         exporter = new NominatimConnector(host, port, database, username, password);
         template = new JdbcTemplate(dataSource);

